website/server/controllers/api-v3/chat.js

```diff
@@ -480,4 +480,58 @@ api.deleteChat = {
   },
 };
 
+/**
+ * @api {post} /api/v3/groups/:groupId/chat/:chatId/approve Approve a group chat message
+ * @apiName ApproveChat
+ * @apiGroup Chat
+ * @apiDescription Approvates a chat message from a group
+ *
+ * @apiParam (Path) {UUID} groupId The group _id ('party' for the user party and 'habitrpg' for tavern are accepted)
+ * @apiParam (Path) {UUID} chatId The chat message _id
+ *
+ * @apiSuccess {Object} data The approved <a href='https://github.com/HabitRPG/habitica/blob/develop/website/server/models/group.js#L51' target='_blank'>chat message</a>
+ *
+ * @apiUse GroupNotFound
+ * @apiUse MessageNotFound
+ * @apiUse GroupIdRequired
+ * @apiUse ChatIdRequired
+ * @apiError (401) {NotAuthorized} adminRequired A user must be an admin to approve a chat message
+ */
+api.approveChat = {
+  method: 'POST',
+  url: '/groups/:groupId/chat/:chatId/approve',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    const user = res.locals.user;
+    const groupId = req.params.groupId;
+
+    req.checkParams('groupId', res.t('groupIdRequired')).notEmpty();
+    req.checkParams('chatId', res.t('chatIdRequired')).notEmpty();
+
+    const validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    if (!user.isAdmin()) throw new NotAuthorized(res.t('noAdminAccess'));
+
+    const group = await Group.getGroup({user, groupId});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    const message = find(group.chat, {id: req.params.chatId});
```

Comment:
This is throwing `ReferenceError: find is not defined` when a mod approves a message.